### ISSUE TITLE
move occurs because `item1.list2` has type `Vec<Box<dyn MyTrait + Send>>`, which does not implement the `Copy` trait

### ISSUE THREAD

**Issue by bbigras** [djc/askama#494]
I got this error with `main` but not with `0.10.5`.

```
error[E0507]: cannot move out of `item1.list2` which is behind a shared reference
 --> src/main.rs:3:10
  |
3 | #[derive(Template)]
  |          ^^^^^^^^ move occurs because `item1.list2` has type `Vec<Box<dyn MyTrait + Send>>`, which does not implement the `Copy` trait
  |
  = note: this error originates in a derive macro (in Nightly builds, run with -Z macro-backtrace for more info)
```

Cargo.toml
```toml
[dependencies]

# works
askama = "0.10.5"

# doesn't work
askama = { git = "https://github.com/djc/askama.git", branch = "main" }
```

main.rs
```rust
use askama::Template;

#[derive(Template)]
#[template(path = "template.txt")]
struct MyStruct1 {
    list1: Vec<MyStruct2>,
}

struct MyStruct2 {
    list2: Vec<Box<dyn MyTrait + Send>>,
}

pub trait MyTrait {
    fn render(&self) -> String;
}

fn main() {}
```

template.txt
```
{% for item1 in list1 %}
{% for item2 in item1.list2 %}
{% endfor %}
{% endfor %}
```

Feel free to rename the issue.

**Comment by vallentin** [comment]
As a workaround, it should work if you change `item1.list2` to `item1.list2.iter()`. However, I'll look into why that fails, as right now I don't recall the reason why it _would_ fail.

**Comment by djc** [comment]
@vallentin would be great if you can look into this, thanks!

**Comment by vallentin** [comment]
@djc I will. I've just been a bit busy the past week and this week, but I've added it to my todo :)

**Comment by bbigras** [comment]
> As a workaround, it should work if you change `item1.list2` to `item1.list2.iter()`.

It works. Thanks :)

**Comment by djc** [comment]
Fixed in #500.
